src/main/java/duke/command/CommandResult.java

```diff
@@ -0,0 +1,43 @@
+package duke.command;
+
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import duke.data.Messages;
+import duke.data.TaskList;
+import duke.data.task.Task;
+
+public class CommandResult {
+    public CommandResult(String top) {
+        this.messageTop = top;
+    }
+
+    public CommandResult(String top, List<Task> target, String bottom) {
+        this.messageTop = top;
+        this.messageBottom = bottom;
+        this.target = target;
+    }
+
+
+    public String[] printTarget() {
+        if (target == null) {
+            return new String[]{};
+        } else if (messageTop.equals(Messages.DIVIDER_LIST)) { /* Case of a List command */
+            return IntStream.range(0, target.size())
+                    .mapToObj(i -> {
```

Comment:
Consider breaking this up so the mapping function is clear?

https://nus-cs2113-ay2223s1.github.io/website/se-book-adapted/chapters/codeQuality.html#practice-kissing